pkg/image/api/helper_test.go

```diff
@@ -741,6 +879,85 @@ func TestImageWithMetadata(t *testing.T) {
 				},
 			},
 		},
+		"valid metadata size": {
+			image: validImageWithManifestV2Data(),
+			expectedImage: Image{
+				ObjectMeta: kapi.ObjectMeta{
+					Name: "id",
+				},
+				DockerImageConfig:            validImageWithManifestV2Data().DockerImageConfig,
+				DockerImageManifest:          validImageWithManifestV2Data().DockerImageManifest,
+				DockerImageManifestMediaType: "application/vnd.docker.container.image.v1+json",
+				DockerImageLayers: []ImageLayer{
+					{Name: "sha256:b4ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4", MediaType: "application/vnd.docker.image.rootfs.diff.tar.gzip", LayerSize: 639152},
+					{Name: "sha256:86e0e091d0da6bde2456dbb48306f3956bbeb2eae1b5b9a43045843f69fe4aaa", MediaType: "application/vnd.docker.image.rootfs.diff.tar.gzip", LayerSize: 235231},
+					{Name: "sha256:86e0e091d0da6bde2456dbb48306f3956bbeb2eae1b5b9a43045843f69fe4aaa", MediaType: "application/vnd.docker.image.rootfs.diff.tar.gzip", LayerSize: 235231},
+					{Name: "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4", MediaType: "application/vnd.docker.image.rootfs.diff.tar.gzip", LayerSize: 5312},
+				},
+				DockerImageMetadata: DockerImage{
+					ID:            "sha256:815d06b56f4138afacd0009b8e3799fcdce79f0507bf8d0588e219b93ab6fd4d",
+					Parent:        "",
+					Comment:       "",
+					Created:       unversioned.Date(2015, 2, 21, 2, 11, 6, 735146646, time.UTC),
+					Container:     "e91032eb0403a61bfe085ff5a5a48e3659e5a6deae9f4d678daa2ae399d5a001",
+					DockerVersion: "1.9.0-dev",
+					Author:        "",
+					Architecture:  "amd64",
+					Size:          882848,
```

Comment:
Have you modified the config in some way? If I count the layer sizes and size of image config as listed in the manifest, I get:
```
639152 + 235231 + 5312 + 7023 = 886718
```

I can get to `882848` if I don't look at image manifest but rather count the size of the config manually.